### PR TITLE
Email message options

### PR DESCRIPTION
**Pull request by kimkeown**
Recently, I sent a pull request pertaining to email_api.php, manage_user_update.php, and bugnote_add_inc.php. For unknown reasons, bugnote_add_inc.php failed the CI check. 

I later received the following response:
https://github.com/mantisbt/mantisbt/pull/902#issuecomment-250029530
> dregad commented 4 hours ago
> @kimkeown thanks for your contribution.
> To facilitate code review, kindly submit one pull request for each distinct change. Also, each pull request should be linked to an issue in our bugtracker. Finally, make sure your submissions do pass the CI check.
> Thanks for your understanding.

 I was unable to find a simple way to create a new pull request, so have deleted the original request to start anew and forward the proposed email revisions.

An initial search did not return a BugTracker issue that specifically addresses my suggestions concerning email message options. However, http://mantisbt.org/bugs/view.php?id=13692 or http://mantisbt.org/bugs/view.php?id=21700 may be applicable. 

Please advise if there are further requirements for review. Thank you!

### CODE REVIEW

**Comment by kimkeown** [comment]
Yesterday CI checks for revised email_api.php and manage_user_update.php passed, and bugnote_add_inc.php failed. 

Today I removed bugnote_add_inc.php, and sent a new pull request for email_api.php and manage_user_update.php. Email_api.php failed the CI check. The failed check files were the last files in queue so perhaps it is due to "one pull request for each distinct change"? The revised files are related in revision, and both would need to be updated to implement the suggested additions.

Kindly advise what might be causing the failed build tests. Instructions or a means of sending separate pull requests would also be appreciated. Thank you!

File: core/email_api.php [hunk 1]
```diff
@@ -1309,8 +1342,14 @@ function email_bug_reminder( $p_recipients, $p_bug_id, $p_message ) {
 			$t_sender_email = '';
 		}
 		$t_header = "\n" . lang_get( 'on_date' ) . ' ' . $t_date . ', ' . $t_sender . ' ' . $t_sender_email . lang_get( 'sent_you_this_reminder_about' ) . ': ' . "\n\n";
-		$t_contents = $t_header . string_get_bug_view_url_with_fqdn( $p_bug_id ) . " \n\n" . $p_message;
 
+/* Kimberly Keown suggested revision to add a custom email signature. (See Line 522.)
+	Revised Code:*/
+	$t_contents = $t_header . string_get_bug_view_url_with_fqdn( $p_bug_id ) . "\n\n" . $p_ message .  "\n\n" . lang_get( 'email_custom_signature' );
```
Comment (dregad) [inline core/email_api.php]: Reason for TravisCI failure. Did you test your code ?

Comment (kimkeown) [inline core/email_api.php]: I'm not sure... does implementing and using the code in my own installation qualify as _testing_??

However, it's true that my bugnote_add_inc.php revision could have been improved to replace the existing class in-line. The email files would require changes to config_defaults_inc.php + language files or custom_strings_inc.php (not included in the current config folder), therefore those additions are referenced only.

As the responses you have posted have been less than constructive, it is clear you are unwilling to fairly review coding that could substantially improve system emails which are sadly lacking in basic customization. That these contributions to an open source program are not welcome is inexplicable. 

My intent in suggesting the revisions is to reciprocate the value I've received from the Bug Tracker, and that remains unchanged. Therefore I am stating my permission to implement the coding I have attempted to provide, at any time now or in the future, without the pull request approval or other acknowledgement.

Thank you for your time.
Kimberly Keown

Comment (dregad) [inline core/email_api.php]: The reason for my earlier comment about testing, is that your code contains a syntax error at the line I pointed out, [as shown in the Travis job log](https://travis-ci.org/mantisbt/mantisbt/jobs/163299393#L201):

```
Parse error: syntax error, unexpected 'message' (T_STRING) in /home/travis/build/mantisbt/mantisbt/core/email_api.php on line 1348
```

(you have a space between `$p_` and `message`).

**Comment by dregad** [comment]
@kimkeown sorry if I came across as not constructive. I am quite busy and had little time to spend for in-depth review and writing detailed posts. I thought it would be useful to give you a quick reply pointing you in a direction that would facilitate future code review process, rather than not responding to your post.

I resent your assertion that I am unwilling to review your code, and your belief that contributions are not welcome is just plain ridiculous.

> I am stating my permission to implement the coding I have attempted to provide, at any time now or in the future, without the pull request approval or other acknowledgement.

I'm not sure I understand what you mean by that. You are of course free to implement whatever you like and as you see fit, but if you want your code to find its way into the product core, you have to abide by our rules. Opening an issue at mantisbt.org is necessary for tracking (think roadmap, changelog) and, optionally, discussing approaches prior to writing code.

**Pull request closed without merging**